test/test_prototype_transforms_consistency.py

```diff
@@ -642,6 +655,38 @@ def test_call_consistency(config, args_kwargs):
     )
 
 
+@pytest.mark.parametrize(
+    ("config", "args_kwargs"),
+    [
+        pytest.param(
+            config, args_kwargs, id=f"{config.legacy_cls.__name__}-{idx:0{len(str(len(config.args_kwargs)))}d}"
+        )
+        for config in CONSISTENCY_CONFIGS
+        for idx, args_kwargs in enumerate(config.args_kwargs)
+        if not isinstance(args_kwargs, NotScriptableArgsKwargs)
+    ],
+)
+def test_jit_consistency(config, args_kwargs):
+    args, kwargs = args_kwargs
+
+    prototype_transform_eager = config.prototype_cls(*args, **kwargs)
+    legacy_transform_eager = config.legacy_cls(*args, **kwargs)
+
+    legacy_transform_scripted = torch.jit.script(legacy_transform_eager)
+    prototype_transform_scripted = torch.jit.script(prototype_transform_eager)
+
+    for image in make_images(**config.make_images_kwargs):
+        image = image.as_subclass(torch.Tensor)
+
+        torch.manual_seed(0)
+        output_legacy_scripted = legacy_transform_scripted(image)
+
+        torch.manual_seed(0)
+        output_prototype_scripted = prototype_transform_scripted(image)
+
+        assert_close(output_prototype_scripted, output_legacy_scripted, **config.closeness_kwargs)
```

Comment:
wondering if the closeness_kwargs actually matters in this case? We should have the exact same results in theory, since we should be comparing the same underlying code?

Reply:
According to @vfdev there were some issues in the past where the eager implementation used Python round and the scripted one the C++ equivalent. However, they are not equivalent in some edge cases which causes flakiness in CI. Given that we have very small tolerances here, I think we should be good as is.

Reply:
you're referring to eager vs scripted where indeed there are minor differences, but this is scripted vs scripted

Reply:
OK, let's just accept JIT is... JIT. Wanna see something **funny**?

```py
import torch
from torchvision import transforms

torch.set_num_threads(1)
scripted = torch.jit.script(transforms.Grayscale())

torch.manual_seed(0)
image = torch.rand(3, 256, 256)

out1 = scripted(image)
out2 = scripted(image)
out3 = scripted(image)

torch.testing.assert_close(out2, out3, atol=0, rtol=0)  # passes
torch.testing.assert_close(out1, out2, atol=0, rtol=0)  # fails
```

Reply:
:smiling_face_with_tear: